lib/Compiler.js

```diff
@@ -121,6 +123,7 @@ Watching.prototype.invalidate = function() {
 	} else {
 		this._go();
 	}
+	callback();
```

Comment:
@kutyel `callback` will not be invoked after the `return` statement https://github.com/webpack/webpack/pull/3103/files#diff-7a14ca420cc4fa7cf55f788246117e7cR122